src/main/host/memory_manager.rs

```diff
@@ -0,0 +1,769 @@
+use super::syscall_types::*;
```

Comment:
Using the wildcard here is a little confusing since there are types with the same names in other modules. For example, this file uses both a `PluginPtr` and a `c::PluginPtr`, which isn't obvious at first glance that they're different types. Rather than the wildcard, I think we should either `use super::syscall_types;` or `use super::syscall_types as <something shorter>`.

Reply:
Okay this makes sense, feel free to leave it as-is.